### PR TITLE
API Refactor: Implements BatchNewX509SVID

### PR DESCRIPTION
**Pull request by MarcosDY**
Implements Batch new x509 svid

**Which issue this PR fixes**
fixes #1566

### CODE REVIEW

File: pkg/server/api/svid/v1/service.go [hunk 1]
```diff
@@ -24,23 +26,36 @@ func RegisterService(s *grpc.Server, service *Service) {
 	svid.RegisterSVIDServer(s, service)
 }
 
+type AuthorizedEntryFetcher interface {
```
[the remark this answers is not in the document]
Comment (MarcosDY) [reply pkg/server/api/svid/v1/service.go:29]: I think singular is fine we already have `registrationEntryFetcher` that allows fetch entries  but getter is plural

File: pkg/server/api/svid/v1/service.go [hunk 2]
```diff
@@ -24,23 +26,36 @@ func RegisterService(s *grpc.Server, service *Service) {
...
+	FetchAuthorizedEntries(ctx context.Context, agentID spiffeid.ID) ([]*types.Entry, error)
+}
+
+type AuthorizedEntryFetcherFunc func(ctx context.Context, agentID spiffeid.ID) ([]*types.Entry, error)
```
[the remark this answers is not in the document]
Comment (MarcosDY) [reply pkg/server/api/svid/v1/service.go:33]: I think singular is fine we already have `registrationEntryFetcher` that allows fetch entries  but getter is plural

**Comment by amartinezfayo** [comment]
`service.go` and `service_test.go` files are getting pretty large and that will become worse as we add more RPCs. I'm wondering if we should split this in `service_x509.go` and `service_jwt.go` files (with its test files each one) and keep `service.go` for shared functions across them?

**Comment by amartinezfayo** [comment]
>service.go and service_test.go files are getting pretty large and that will become worse as we add more RPCs. I'm wondering if we should split this in service_x509.go and service_jwt.go files (with its test files each one) and keep service.go for shared functions across them?

@azdagron do you have any thoughts on this? maybe not for this PR but would be something worth considering?

**Comment by azdagron** [comment]
@amartinezfayo I've been thinking about this. There are some nice things about having it all in one file but I'd be open to breaking things up, as long as we're consistent. For example, doing "topic" breakdown (i.e. between JWT and X509) works here because there are just a few methods for each SVID type, but such a break down wouldn't work well for the Bundle API for example.

What about one-file-per-RPC approach with some consistent naming pattern, like `rpc_mint_x509svid.go`, `rpc_mint_downstream_ca`, etc.?

**Comment by MarcosDY** [comment]
I was thinking about that and it will result in more files. 
In case we split we'll need a `commont_test` where we put all methods in common like testService, and there will be dependencies there.

**Comment by amartinezfayo** [comment]
>What about one-file-per-RPC approach with some consistent naming pattern, like rpc_mint_x509svid.go, rpc_mint_downstream_ca, etc.?

This looks good to me. I think that in any case we can see how this evolve after merging these RPCs and look at the split strategy once we can see what are the shared functions, etc.

File: pkg/server/api/svid/v1/service.go [hunk 3]
```diff
@@ -231,37 +231,37 @@ func (s *Service) newX509SVID(ctx context.Context, param *svid.NewX509SVIDParams
...
 	case param.EntryId == "":
-		log.Error("Invalid param: missing Entry ID")
-		return nil, status.Error(codes.InvalidArgument, "invalid param: missing Entry ID")
+		log.Error("Invalid request: missing entry ID")
+		return nil, status.Error(codes.InvalidArgument, "missing entry ID")
 	case len(param.Csr) == 0:
-		log.WithField(telemetry.RegistrationID, param.EntryId).Error("Invalid param: missing CSR")
-		return nil, status.Errorf(codes.InvalidArgument, "invalid param: entry %q missing CSR", param.EntryId)
+		log.Error("Invalid request: missing CSR")
+		return nil, status.Errorf(codes.InvalidArgument, "missing CSR")
```
Comment (azdagron) [inline pkg/server/api/svid/v1/service.go:238]:
```suggestion
		return nil, status.Error(codes.InvalidArgument, "missing CSR")
```

**Pull request merged**